pkg/forth/forth.go

```diff
@@ -157,39 +185,62 @@ func errRecover(errp *error) {
 	}
 }
 
-/* iEval takes a Forth and strings and splits the string on space
- * characters, pushing each element on the stack or invoking the
- * operator if it is found in the opmap.
- */
-func iEval(f Forth, s string) {
-	// TODO: create a separator list based on isspace and all the
-	// non alpha numberic characters in the opmap.
-	for _, val := range strings.Fields(s) {
-		//fmt.Printf("eval %s stack %v", val, f.Stack())
-		fun := opmap[val]
-		if fun != nil {
-			//fmt.Printf("Eval ...:")
-			fun(f)
-			//fmt.Printf("Stack now %v", f.Stack())
-		} else {
-			f.Push(val)
-			//fmt.Printf("push %s, stack %v", val, f.Stack())
+// Eval takes a Forth and []Cell, pushing each element on the stack or invoking the
+// operator if it is found in the opmap.
+func Eval(f Forth, cells ...Cell) (err error) {
+	defer errRecover(&err)
+	for _, c := range cells {
+		Debug("eval %s(%T) stack %v", c, c, f.Stack())
+		switch s := c.(type) {
+		case string:
+			fun := Getop(s)
+			if fun != nil {
+				Debug("Eval ... %v:", f.Stack())
+				fun(f)
+				Debug("Stack now %v", f.Stack())
+			} else {
+				f.Push(s)
+				Debug("push %s, stack %v", s, f.Stack())
+			}
+		default:
+			Debug("push %s, stack %v", s, f.Stack())
+			f.Push(s)
+		}
+	}
+	return nil
+}
+
+// EvalString takes a Forth and string and splits the string on space
+// characters, calling Eval for each one.
+func EvalString(f Forth, s string) (err error) {
+	defer errRecover(&err)
```

Comment:
Is this `defer` necessary? The `Eval` function it calls has the same defer.

Reply:
removed. Thanks.